File: src/main/java/com/nibado/projects/advent/y2015/AdventOfCode.java
package com.nibado.projects.advent.y2015;


import java.util.Arrays;
import java.util.List;

public class AdventOfCode {
    public static void main(String... argv) {
        List<Runnable> solutions = Arrays.asList(
                new Day01J(),
                new Day02J(),
                new Day03J(),
                new Day04J(),
                new Day05J(),
                new Day06J(),
                new Day07J(),
                new Day08J(),
                new Day09J(),
                new Day10J(),
                new Day11J(),
                new Day12J(),
                new Day13J(),
                new Day14J(),
                new Day15J(),
                new Day16J());

        solutions.forEach(Runnable::run);
    }
}

File: src/main/kotlin/com/nibado/projects/advent/y2015/AdventOfCode.kt
package com.nibado.projects.advent.y2015

import com.nibado.projects.advent.Reflect
import com.nibado.projects.advent.Runner

object AoC2015 {
    val days = Reflect.getDays(2015)
}

fun main(args: Array<String>) {
    val day = if(args.isEmpty()) 0 else args[0].toInt()

    Runner.run(2015, AoC2015.days, day)
}